pkg/operator/ceph/object/objectstore.go

```diff
@@ -1000,3 +999,38 @@ func errorOrIsNotFound(err error, msg string, args ...string) error {
 	}
 	return errors.Wrapf(err, msg, args)
 }
+
+// check whether zone endpoint list need to be updated or not
+func UpdateZoneEndpointList(zones []zoneType, desiredEndpointList []string, zoneName string) bool {
+	updateZoneEndpointList := false
+	isZonePresent := false
+	zoneEndpointsList := []string{}
+	checkZoneEndpoints := make(map[string]bool)
+	if zoneName != "" {
+		if len(zones) > 0 {
+			for _, z := range zones {
+				if z.Name == zoneName {
+					for _, endpoint := range z.Endpoints {
+						zoneEndpointsList = append(zoneEndpointsList, endpoint)
+						checkZoneEndpoints[endpoint] = true
+					}
+					isZonePresent = true
```

Comment:
It would be better to write a helper function that returns whether the zone is present and the endpoint list of said zone.

```
func findZoneEndpoints(targetZone string, zones []zoneType{}) (bool, []string) {
    for _, z := range zones {
        if z.Name == targetZone {
            return true, z.Endpoints
        }
    }
    return false, []string{}
} 
```